build/source/engine/getVectorz.f90

```diff
@@ -149,68 +194,137 @@ real(dp),intent(out)            :: fScale(:)              ! function scaling vec
+ end do
+
+ ! build the state vector for the temperature of the vegetation canopy
+ ! NOTE: currently vector length=1, and use "do concurrent" to generalize to a multi-layer canopy
+ do concurrent (iState=1:size(ixVegNrg),ixVegNrg(iState)/=integerMissing)
+  stateVec( ixVegNrg(iState) )  = scalarCanopyTemp       ! transfer vegetation temperature to the state vector
+  stateFlag( ixVegNrg(iState) ) = .true.                 ! flag to denote that the state is populated
+ end do
+
+ ! build the state vector for the water in the vegetation canopy
+ ! NOTE: currently vector length=1, and use "do concurrent" to generalize to a multi-layer canopy
+ do concurrent (iState=1:size(ixVegHyd),ixVegHyd(iState)/=integerMissing)
+  select case(ixStateType_subset( ixVegHyd(iState) ))
+   case(iname_watCanopy); stateVec( ixVegHyd(iState) )  = scalarCanopyWat        ! transfer total canopy water to the state vector
+   case(iname_liqCanopy); stateVec( ixVegHyd(iState) )  = scalarCanopyLiq        ! transfer liquid canopy water to the state vector
+   case default; cycle
+  end select
+  stateFlag( ixVegHyd(iState) ) = .true.                 ! flag to denote that the state is populated
+ end do
+
+ ! build the energy state vector for the snow and soil domain
+ if(nSnowSoilNrg>0)then
+  do concurrent (iLayer=1:nLayers,ixSnowSoilNrg(iLayer)/=integerMissing)   ! (loop through non-missing energy state variables in the snow+soil domain)
+   ixStateSubset            = ixSnowSoilNrg(iLayer)  ! index within the state vector
+   stateVec(ixStateSubset)  = mLayerTemp(iLayer)     ! transfer temperature from a layer to the state vector
+   stateFlag(ixStateSubset) = .true.                 ! flag to denote that the state is populated
+  end do  ! looping through non-missing energy state variables in the snow+soil domain
+ endif
+
+ ! build the hydrology state vector for the snow+soil domains
+ ! NOTE: ixVolFracWat  and ixVolFracLiq can also include states in the soil domain, hence enable primary variable switching
+ if(nSnowSoilHyd>0)then
+  do concurrent (iLayer=1:nLayers,ixSnowSoilHyd(iLayer)/=integerMissing)   ! (loop through non-missing hydrology state variables in the snow+soil domain)
+   ixStateSubset                                  = ixSnowSoilHyd(iLayer)   ! index within the state vector
+   select case( ixHydType(iLayer) )
+    case(iname_watLayer); stateVec(ixStateSubset) = mLayerVolFracWat(iLayer)           ! total water state variable for snow+soil layers
+    case(iname_liqLayer); stateVec(ixStateSubset) = mLayerVolFracLiq(iLayer)           ! liquid water state variable for snow+soil layers
+    case(iname_matLayer); stateVec(ixStateSubset) = mLayerMatricHead(iLayer-nSnow)     ! total water matric potential variable for soil layers
+    case(iname_lmpLayer); stateVec(ixStateSubset) = mLayerMatricHeadLiq(iLayer-nSnow)  ! liquid matric potential state variable for soil layers
+    case default; cycle
```

Comment:
These `cycle` from concurrent do loops do not work in ifort 14.X.X:
`A CYCLE or EXIT statement must not be used with a non-DO block`

The ifort documentation says that cycle will work with concurrent do loops: https://software.intel.com/en-us/node/526121

There is a new 2008 standard about exiting from case selects using `exit` (http://stackoverflow.com/questions/25676793/break-from-select-case), and I wonder if they have defaulted the behavior of the compiler to check for `cycles` instead of `exits` from case selects. 

Either way, we will probably want to change this to a more universal approach - for example, using an if block around the `stateFlag` assignment.